components/payments.tsx

```diff
@@ -1,58 +1,137 @@
-import { Badge, Flex, Box, Card, IconButton, Paragraph, Link } from '@theme-ui/components';
+import type { Stripe_PaymentIntent, Stripe_Invoice, Stripe_Item, Stripe_LineItem } from 'lib/takeshape/types';
+import NextLink from 'next/link';
+import { Badge, Flex, Box, Card, IconButton, Paragraph, Link, Heading, Text, Grid } from '@theme-ui/components';
 import { formatPrice } from 'lib/utils/text';
 import { format } from 'date-fns';
 import startCase from 'lodash/startCase';
 import { BsCheckCircleFill, BsReceipt } from 'react-icons/bs';
-
+import { ProductImage } from './product';
 export interface OrderStatusProps {
-  status: 'processing' | 'shipped' | 'delivered';
+  status: 'unknown' | 'in_transit' | 'delivered' | 'error';
   trackingNumber?: string | null;
 }
 
-export const OrderStatus: React.FC<OrderStatusProps> = ({ status, trackingNumber }) => {
-  const colorMap = {
-    processing: 'primary',
-    shipped: 'gray',
-    delivered: 'purple'
+export const OrderStatus = ({ status, trackingNumber }: OrderStatusProps) => {
+  const statusMap = {
+    unknown: {
+      text: 'Processing',
+      color: 'gray'
+    },
+    in_transit: {
+      text: 'Shipped',
+      color: 'green'
+    },
+    delivered: {
+      text: 'Delivered',
+      color: 'purple'
+    },
+    error: {
+      text: 'Error',
+      color: 'red'
+    }
   };
+
+  const statusObj = statusMap[status];
+
   return (
     <Flex sx={{ gap: '.5rem' }}>
-      <Badge py={1} px={2} backgroundColor={colorMap[status] ?? 'gray'}>
-        {startCase(status)}
+      <Badge py={1} px={2} backgroundColor={statusObj.color}>
+        {startCase(statusObj.text)}
```

Comment:
I don't think we need the startCase formatting any more, since that's being set within the object.

```suggestion
        {statusObj.text}
```